src/zarr/api/asynchronous.py

```diff
@@ -894,9 +921,13 @@ async def create(
         or _default_zarr_version()
     )
 
-    if zarr_format == 2 and chunks is None:
-        chunks = shape
-    elif zarr_format == 3 and chunk_shape is None:
+    if zarr_format == 2:
+        if chunks is None:
+            chunks = shape
+        dtype = parse_dtype(dtype, zarr_format)
+        if not filters and not compressor:
+            filters, compressor = _default_filters_and_compressor(dtype)
+    elif zarr_format == 3 and chunk_shape is None:  # type: ignore[redundant-expr]
```

Comment:
```suggestion
    elif chunk_shape is None:
```
Since by this point zarr_format is definitely either 2 or 3. My suggestion is a bit less readable though, so feel free to take or leave.

Reply:
I had the same thought and preferred having `zarr_format == 3` for readability